### PR TITLE
Chore: use correct ci dependency versions

### PR DESCRIPTION
**Pull request by ThisIsMissEm**
This addresses a lot of the open dependabot PRs.

closes #235 #234 #230 #207

### CODE REVIEW

File: .github/workflows/e2e-node.yml [hunk 1]
```diff
@@ -22,50 +22,50 @@ jobs:
...
+        env:
+          E2E_TEST_NOTIFICATION_GATEWAY: ${{ secrets.E2E_TEST_NOTIFICATION_GATEWAY }}
+          E2E_TEST_NOTIFICATION_PROTOCOL: ${{ secrets.E2E_TEST_NOTIFICATION_PROTOCOL }}
+          E2E_TEST_IDP: ${{ secrets.E2E_TEST_IDP }}
+          E2E_TEST_CLIENT_ID: ${{ secrets.E2E_TEST_CLIENT_ID }}
+          E2E_TEST_CLIENT_SECRET: ${{ secrets.E2E_TEST_CLIENT_SECRET }}
+          E2E_TEST_ENVIRONMENT: ${{ matrix.environment-name }}
+      - name: Add status check with successful test result
+        id: set-status-check-success
+        uses: octokit/request-action@v2
```
Comment (NSeydoux) [inline .github/workflows/e2e-node.yml:43]: It looks like this is problematic. Should the patch version be still specified, and if so do you have an idea why it differs from other GH actions?

Comment (ThisIsMissEm) [reply .github/workflows/e2e-node.yml:43]: Well, our current directive is that we don't care for patch versions. Though, honestly, maybe we can get rid of this? idk

Comment (ThisIsMissEm) [reply .github/workflows/e2e-node.yml:43]: Apparently it should be
```suggestion
        uses: octokit/request-action@v2.x
```

Comment (NSeydoux) [reply .github/workflows/e2e-node.yml]: I saw that in the docs, but I thought it was a hand-wavy example, and not the actual expected value ^^

**Review by NSeydoux — APPROVED** [review]

**Pull request merged**